server.go

```diff
@@ -1596,20 +1596,26 @@ func (s *Server) clearExpiredClients(dt int64) {
 
 // clearExpiredRetainedMessage deletes retained messages from topics if they have expired.
 func (s *Server) clearExpiredRetainedMessages(now int64) {
-	for filter, pk := range s.Topics.Retained.GetAll() {
-		if (pk.Expiry > 0 && pk.Expiry < now) || pk.Created+s.Options.Capabilities.MaximumMessageExpiryInterval < now {
-			s.Topics.Retained.Delete(filter)
-			s.hooks.OnRetainedExpired(filter)
+	// If the maximum message expiry interval is set to math.MaxInt64, do not process expired messages.
+	if s.Options.Capabilities.MaximumMessageExpiryInterval != math.MaxInt64 {
```

Comment:
If we were to implement this change, and I'm not certain we should, this may be more in line with the existing style:

```suggestion
	if s.Options.Capabilities.MaximumMessageExpiryInterval == math.MaxInt64 {
	    return
	}
	
	for filter, pk := range s.Topics.Retained.GetAll() {
	
```